### DISCUSSION TITLE
Link two different jobs?

### DISCUSSION THREAD

**Discussion by engineertdog** [quartznet/quartznet#2228] [Q&A]
I have two different classes that extract data and store it in a database. However, the two data sets from each class are tied together and I need to find a way to link them. 

1. One of the ways to link them would be to merge the two classes (and SQL tables). But while the data sets are related, they do still serve a purpose as separate entities and should be treated as such.
2. I could introduce a 3rd new class to handle the joined data.
3. Use the job data to generate some identifier to link the classes. The jobs are going to be executed using the same cron schedule, say 15 seconds. I thought about either using an incrementing counter, or using the date of execution as a means to link the two runs, but the milliseconds on the execution of the jobs are not always going to be in alignment.

```c#
public virtual async Task Execute(IJobExecutionContext context) {
  JobDataMap dataMap = context.MergedJobDataMap;
  Log.Warning(DateTime.Now.Millisecond.ToString()); // not the same for both classes

  long count = dataMap.GetLong("jobIterationNumber");
  dataMap.Put("jobIterationNumber", ++count); // does not work like this
  ...
}
```

Option 2 might be the best (but it does duplicate data - and the two classes will generate a lot of records), but I was curious on other's inputs.

**Comment by lahma** [comment]
If you want two jobs running idenpendently sharing processing ID you could just create one trigger against the suggested 3rd job which has responsibility to generate the id (either using its own fire instance id or generated guid maybe) and it would call context's scheduler's TtiggerJob for the two job types with job data map containing the generated id.

So one job and trigger configured, and this job triggers the two jobs manually.
